### PR TITLE
chore(Core/ScriptMgr): Hooks used in mod-aoe-loot

### PR DESCRIPTION
**Pull request by pangolp**
## Changes Proposed:

- Hook CanSendErrorAlreadyLooted
- Hook OnAfterCreatureLoot
- Hook OnAfterCreatureLootMoney

## Issues Addressed:

- Closes 

## Tests Performed:

- In-game
- Windows 10

## How to Test the Changes:

1. The truth is that it is difficult to test the pull request without the module as such, but the module does not work if we do not have the 3 hooks in the emulator.

## Known Issues and TODO List:

https://github.com/azerothcore/azerothcore-wotlk/assets/2810187/fdf35a3a-bc6a-447f-91bd-2f52732b6f4e

https://github.com/azerothcore/azerothcore-wotlk/assets/2810187/24dd6685-8e9f-4980-83f6-518018412cc6

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -13442,7 +13442,11 @@ LootItem* Player::StoreLootItem(uint8 lootSlot, Loot* loot, InventoryResult& msg
     LootItem* item = loot->LootItemInSlot(lootSlot, this, &qitem, &ffaitem, &conditem);
     if (!item || item->is_looted)
     {
-        SendEquipError(EQUIP_ERR_ALREADY_LOOTED, nullptr, nullptr);
+        Player* player = nullptr;
+        if (!sScriptMgr->CanSendErrorArleadyLooted(player))
```
Comment (0xCiBeR) [inline src/server/game/Entities/Player/Player.cpp:13446]: Typo? I think this should be called CanSendErrorAlreadyLooted

File: src/server/game/Scripting/ScriptDefines/PlayerScript.cpp [hunk 2]
```diff
@@ -905,6 +905,44 @@ bool ScriptMgr::CanSendMail(Player* player, ObjectGuid receiverGuid, ObjectGuid
     return true;
 }
 
+bool ScriptMgr::CanSendErrorArleadyLooted(Player* player)
+{
+    auto ret = IsValidBoolScript<PlayerScript>([&](PlayerScript* script)
+    {
+        return !script->CanSendErrorArleadyLooted(player);
```
Comment (0xCiBeR) [inline src/server/game/Scripting/ScriptDefines/PlayerScript.cpp:912]: Typo? I think this should be called CanSendErrorAlreadyLooted

File: src/server/game/Scripting/ScriptDefines/PlayerScript.cpp [hunk 3]
```diff
@@ -905,6 +905,44 @@ bool ScriptMgr::CanSendMail(Player* player, ObjectGuid receiverGuid, ObjectGuid
     return true;
 }
 
+bool ScriptMgr::CanSendErrorArleadyLooted(Player* player)
```
Comment (0xCiBeR) [inline src/server/game/Scripting/ScriptDefines/PlayerScript.cpp:908]: Typo? I think this should be called CanSendErrorAlreadyLooted

File: src/server/game/Scripting/ScriptMgr.h [hunk 4]
```diff
@@ -1458,6 +1458,11 @@ class PlayerScript : public ScriptObject
     virtual void AnticheatUpdateMovementInfo(Player* /*player*/, MovementInfo const& /*movementInfo*/) { }
     [[nodiscard]] virtual bool AnticheatHandleDoubleJump(Player* /*player*/, Unit* /*mover*/) { return true; }
     [[nodiscard]] virtual bool AnticheatCheckMovementInfo(Player* /*player*/, MovementInfo const& /*movementInfo*/, Unit* /*mover*/, bool /*jump*/) { return true; }
+
+    virtual bool CanSendErrorArleadyLooted(Player* /*player*/) { return true; }
```
Comment (0xCiBeR) [inline src/server/game/Scripting/ScriptMgr.h:1462]: Typo? I think this should be called CanSendErrorAlreadyLooted

File: src/server/game/Scripting/ScriptMgr.h [hunk 5]
```diff
@@ -2381,6 +2386,9 @@ class ScriptMgr
     void OnPlayerEnterCombat(Player* player, Unit* enemy);
     void OnPlayerLeaveCombat(Player* player);
     void OnQuestAbandon(Player* player, uint32 questId);
+    bool CanSendErrorArleadyLooted(Player* player);
```
Comment (0xCiBeR) [inline src/server/game/Scripting/ScriptMgr.h:2389]: Typo? I think this should be called CanSendErrorAlreadyLooted

**Comment by pangolp** [comment]
@Nyeriah Could you please run the tests again? Something went wrong, and I don't know what it is. I think it has nothing to do with the code, but just in case. Thanks.

**Comment by pangolp** [comment]
If someone can review and merge this PR, I would appreciate it, so I can finish making the loot mod in area. Unless of course, some changes need to be made to it, so I need to know what to change. Maybe the documentation is too little, or the names are not very representative. I don't know, any change, in order to improve, is welcome.

Pd. I know you are busy with other pull request, and maybe this is not something really relevant for the emulator, however, the module requires this change to start working and receive the comments from the community.

Thank you.

**Review by 0xCiBeR — APPROVED** [review]

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**